test/functional/test_framework/test_framework.py

```diff
@@ -170,6 +170,8 @@ def parse_args(self):
         parser.add_argument("--descriptors", default=False, action="store_true",
                             help="Run test using a descriptor wallet")
         parser.add_argument('--factor', type=float, default=1.0, help='adjust test timeouts by a factor')
+        parser.add_argument("--notimeout", dest="notimeout", default=False, action="store_true",
+                            help="Disable the test timeout, should not be used with --factor")
```

Comment:
why not make any negative factor map to factor=inf?

Reply:
(ofc with proper documentation)

Reply:
i like the simplicity of this

Reply:
That definitely can be done, but I feel it would unnecessarily complicate it for new testers. It won't be readily obvious unless they look it up. 
Also, I have issues with factor doc, it wasn't clear that it related to rpc_timeout until I look it up. (should it be changed?) 
Lastly I feel keeping the --notimeout flag explicit, helps differentiate the purpose of two.

Reply:
>  it wasn't clear that it related to rpc_timeout 

then that sounds like a documentation issue that should be fixed, not worked around

Reply:
I hope its clear that the work is not around --factor. I have stated that the similar effect can be achieved by just using a higher factor. The addition only helps to reduce the burden of figuring it out. There are people who are setting env variables and manually changing the code to get this behaviour. Having a flag only simplifies that issue.